src/shaders/pbr.frag

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+// This is an implementation of
+// Karis, Brian. “Real Shading in Unreal Engine 4 by.” (2013).
+
+// -------------- input ---------------------
+// position, normal, tangent in camera space
+in highp vec3 position;
+in highp vec3 normal;
+#if defined(TEXTURED)
+in highp vec2 texCoord;
+#endif
+#if defined(NORMAL_TEXTURE) && defined(PRECOMPUTED_TANGENT)
+in highp vec3 tangent;
+in highp vec3 biTangent;
+#endif
+
+// -------------- output -------------------
+layout(location = OUTPUT_ATTRIBUTE_LOCATION_COLOR) out vec4 fragmentColor;
+#if defined(OBJECT_ID)
+layout(location = OUTPUT_ATTRIBUTE_LOCATION_OBJECT_ID) out highp uint
+    fragmentObjectId;
+#endif
+
+#if (LIGHT_COUNT > 0)
+// -------------- lights -------------------
+// NOTE: In this shader, the light intensity is considered in the lightColor!!
+uniform vec3 LightColors[LIGHT_COUNT];
+uniform float LightRanges[LIGHT_COUNT];
+
+// if .w == 0, it means it is a dirctional light, .xyz is the direction;
+// if .w == 1, it means it is a point light, .xyz is the light position;
+// it is NOT put in the Light Structure, simply because we may modify the code
+// so it is computed in the vertex shader.
+uniform vec4 LightDirections[LIGHT_COUNT];
+#endif
+
+// -------------- material, textures ------------------
+struct MaterialData {
+  vec4 baseColor;     // diffuse color, if baseColorTexture exists,
```

Comment:
Does the alpha component mean anything?

Reply:
Yep, glTF supports alpha-masked and alpha-blended materials.

Reply:
It means a lot.
We need the alpha value if the object is a transparent or semi-transparent object.